tools/helper_builder.rb

```diff
@@ -85,15 +88,39 @@ def go_available?
     end
   end
 
+  def buildable_archs
+    if @archs.nil?
+      version = run_go_version[/go(\d+.\d)/, 1].to_f
+      @archs = if version <= 1.4
+        case run_uname_p
+        when "x86_64"
+          ["x86_64"]
+        when "i686"
+          ["i686"]
+        else
+          []
+        end
+      elsif version <= 1.6
+        archs = ["i686", "x86_64", "ppc64le"]
+        # the two following lines are temporary until Go 1.7 with s390x support is released
+        archs.push("s390x") if File.exist?("/usr/share/go/src/cmd/asm/internal/arch/s390x.go")
+        archs
+      else
+        ["i686", "x86_64", "ppc64le", "s390x"]
+      end
+    end
+    @archs
+  end
+
   def arch_supported?
     arch = run_uname_p
-    if !["x86_64"].include?(arch)
+    if buildable_archs.include?(arch)
```

Comment:
In general all these specific checks sound like they would suit better to be handled by a different class so we can reduce the amount of nested helps and having all the logic in one place